sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ApproximatePercentile.scala

```diff
@@ -226,7 +226,7 @@ object ApproximatePercentile {
    *
    * @param summaries underlying probabilistic data structure [[QuantileSummaries]].
    */
-  class PercentileDigest(private var summaries: QuantileSummaries) {
+  class PercentileDigest(private var summaries: QuantileSummaries) extends Serializable {
```

Comment:
hmm, there are special methods for (de-)serialization of `PercentileDigest`:
https://github.com/apache/spark/blob/08123a3795683238352e5bf55452de381349fdd9/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ApproximatePercentile.scala#L212-L218
Do you know why the methods are used instead of extending of `Serializable`?

cc @cloud-fan

Reply:
Can we just add the proper logic to observe? Adding serialization will fix this particular problem, but it won't work for other aggregates.

Reply:
I agree if that's possible - I couldn't tell if this was something observe() should never have to pull back and does accidentally, or whether it is necessary to return metrics about percentile_approx somehow

Reply:
I probably forgot to add it...

Reply:
@hvanhovell I updated the code and it will fix the issue for `AggregatingAccumulator` takes any `TypedImperativeAggregate`.